### ISSUE TITLE
CA1806 never produces a diagnostic

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn-analyzers#1845]
#### Analyzer package

[Microsoft.Maintainability.Analyzers](https://www.nuget.org/packages/Microsoft.Maintainability.Analyzers)

#### Package Version

v1.2.0-beta2

#### Diagnostic ID

[CA1806](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1806-do-not-ignore-method-results?view=vs-2015)

#### Repro steps

```cs
class Program
{
    [Pure]
    static int Foo() => 5;
  
    static void Main()
    {
        Foo();
    }
}
```

#### Expected behavior

A CA1806 diagnostic produced on the invocation of `Foo()`.

#### Actual behavior

No diagnostic.

While `[Pure]` methods aren't explicitly called out in the documentation for this rule, I believe it should be considered just like the other scenarios that are listed. It looks like the `[Pure]` support is tracked by https://github.com/dotnet/roslyn-analyzers/issues/746#issuecomment-163084601

I tried some of the other ones listed in the docs and they too did not produce any diagnostic.

**Comment by mavasani** [comment]
Does not repro with latest stable version (v2.6.2) of the analyzer package.

![image](https://user-images.githubusercontent.com/10605811/50573951-a5d38580-0d92-11e9-83c1-62100c5fd103.png)

**Comment by AArnott** [comment]
So I guess `Microsoft.Maintainability.Analyzers` is a deprecated package and I should switch to `Microsoft.CodeQuality.Analyzers`. Thanks.

**Comment by mavasani** [comment]
Ah, I didn't notice you had mentioned `Microsoft.Maintainability.Analyzers` in the repro. Yes, that package has been deprecated since we refactored the analyzer packages for the netcore/netstandard era.
